fe/fe-core/src/main/java/com/starrocks/sql/optimizer/rule/transformation/materialization/AggregatedMaterializedViewRewriter.java

```diff
@@ -0,0 +1,462 @@
+    }
+
+    private Map<ColumnRefOperator, CallOperator> rewriteAggregates(Map<ColumnRefOperator, ScalarOperator> aggregates,
+                                                           Multimap<ScalarOperator, ColumnRefOperator> normalizedViewMap,
+                                                           Map<ColumnRefOperator, ColumnRefOperator> mapping,
+                                                           Set<ColumnRefOperator> queryColumnSet) {
+        Map<ColumnRefOperator, CallOperator> rewrittens = Maps.newHashMap();
+        for (Map.Entry<ColumnRefOperator, ScalarOperator> entry : aggregates.entrySet()) {
+            Preconditions.checkState(entry.getValue() instanceof CallOperator);
+            CallOperator aggCall = (CallOperator) entry.getValue();
+            ScalarOperator targetColumn = replaceExprWithTarget(aggCall, normalizedViewMap, mapping);
+            if (!isAllExprReplaced(targetColumn, queryColumnSet)) {
+                // it means there is some column that can not be rewritten by outputs of mv
+                return null;
+            }
+            // Aggregate must be CallOperator
+            Preconditions.checkState(targetColumn instanceof ColumnRefOperator);
+            CallOperator newAggregate = getRollupAggregate(aggCall, (ColumnRefOperator) targetColumn);
+            if (newAggregate == null) {
+                return null;
+            }
+            rewrittens.put(entry.getKey(), newAggregate);
+        }
+
+        return rewrittens;
+    }
+
+    private Map<ColumnRefOperator, CallOperator> rewriteAggregatesForUnion(Map<ColumnRefOperator, CallOperator> aggregates,
+                                                                           Map<ColumnRefOperator, ColumnRefOperator> mapping) {
+        Map<ColumnRefOperator, CallOperator> rewrittens = Maps.newHashMap();
+        for (Map.Entry<ColumnRefOperator, CallOperator> entry : aggregates.entrySet()) {
+            Preconditions.checkState(entry.getValue() instanceof CallOperator);
+            CallOperator aggCall = entry.getValue();
+            ColumnRefOperator targetColumn = mapping.get(entry.getKey());
+            // Aggregate must be CallOperator
+            Preconditions.checkState(targetColumn instanceof ColumnRefOperator);
+            CallOperator newAggregate = getRollupAggregate(aggCall, targetColumn);
+            if (newAggregate == null) {
+                return null;
+            }
+            rewrittens.put(entry.getKey(), newAggregate);
+        }
+
+        return rewrittens;
+    }
+
+    // generate new aggregates for rollup
+    // eg: count(col) -> sum(col)
+    private CallOperator getRollupAggregate(CallOperator aggCall, ColumnRefOperator targetColumn) {
+        if (ROLLUP_FUNCTION_MAP.containsKey(aggCall.getFnName())) {
+            if (aggCall.getFnName().equals(FunctionSet.COUNT)) {
+                Function sumFn = findArithmeticFunction(aggCall.getFunction().getArgs(), FunctionSet.SUM);
```

Comment:
ROLLUP_FUNCTION_MAP only contains (count -> sum). Why use both containsKey and equals check?